src/Microsoft.OData.Core/UriParser/Parsers/ODataPathParser.cs

```diff
@@ -449,17 +449,145 @@ private bool TryCreateCountSegment(string segmentText)
             return true;
         }
 
+        /// <summary>
+        /// Retrieves the last navigation source and index of the navigation source in the parsed segments.
+        /// </summary>
+        /// <param name="navigationSource">Extracted navigation source. This is null if no source is found.</param>
+        /// <param name="navigationSourceIndex">Extracted index of the navigation source. This is -1 if no source is found.</param>
+        /// <returns>True if a navigation sourse was found; false, otherwise.</returns>
+        private bool GetLastNavigationSource(out IEdmNavigationSource navigationSource, out int navigationSourceIndex)

```

Comment:
Can we assume that the previous segment has the navigationSource info?

Reply:
OperationSegments currently do not necessarily have navigationSources, so we cannot assume the previous segment to have navigationSource info. Going to remove this method and instead add a private member to store the last parsed navigationSource.